crates/macros/src/syn_ext.rs

```diff
@@ -136,6 +136,7 @@ impl DropLifetimes for syn::TypePath {
         if let Some(qself) = &mut self.qself {
             qself.ty.drop_lifetimes();
         }
+        self.path.segments.drop_lifetimes();
```

Comment:
Tell me if I'm wrong but if this is the part that removes lifetimes from items, wouldn't it be better to throw an error here ?
Because currently if you try to add a lifetime to a registered class, this is the compiler message:
```
error[E0726]: implicit elided lifetime not allowed here
 --> src/lib.rs:4:8
  |
4 | struct Foo<'a> {
  |        ^^^ expected lifetime parameter
  |
  = note: assuming a `'static` lifetime...
help: indicate the anonymous lifetime
  |
4 | struct Foo<'_><'a> {
  |           ++++
```
Which is rather misleading.
Given that the restrictions on lifetime and generic parameters are easily overlooked by newcomers, I think that an error stating explicitely that this is not allowed by the library would be better.

Reply:
Answering my past self: this part drops lifetimes of function and method arguments.

Since I don't agree with the type conversion magic at function boundaries, I have no opinion on whether dropping lifetime for arguments is a good thing or not.

But I still think that `#[php_class]` should fail to compile with an error message like
````
Lifetimes and generic parameters are not allowed on classes and class members
```

instead of the message seen in the [previous comment](https://github.com/davidcole1340/ext-php-rs/pull/174/files#diff-d69bdb20c8c49926e0b3f843284710b84915fb209e97d120290136743587607d).